### PR TITLE
refactor(api): AbstractInstrument.aspirate()

### PR DESCRIPTION
**Pull request by SyntaxColoring**

### CODE REVIEW

File: api/src/opentrons/protocol_api/instrument_context.py [hunk 1]
```diff
@@ -184,37 +184,14 @@ def aspirate(self,
                 " method that moves to a location (such as move_to or "
                 "dispense) must previously have been called so the robot "
                 "knows where it is.")
-
-        if self.current_volume == 0:
```
Comment (amitlissack) [inline api/src/opentrons/protocol_api/instrument_context.py:188]: I had hoped that this block of code could go in whole to  `api/src/opentrons/protocols/context/protocol_api/instrument_context.py`

File: api/src/opentrons/protocols/context/protocol_api/instrument_context.py [hunk 2]
```diff
@@ -57,12 +57,64 @@ def set_default_speed(self, speed: float) -> None:
...
+        c_vol = self.get_available_volume() \
+            if not volume else volume
+
+        # FIX BEFORE MERGE: I don't understand what these do_publish things
+        # are, but we presumably need to retain them.
+        #
+        # Give this AbstractInstrument a broker when it's constructed?
+        #
+        # What does "self.aspirate" mean here? Is it safe to change to
+        # "self.aspirate_in_place"?
```
Comment (amitlissack) [inline api/src/opentrons/protocols/context/protocol_api/instrument_context.py:112]: It is not safe to change that. `do_publish` is the tie in to the run log.

File: api/src/opentrons/protocols/context/protocol_api/instrument_context.py [hunk 3]
```diff
@@ -57,12 +57,64 @@ def set_default_speed(self, speed: float) -> None:
...
+        do_publish(self.broker, cmds.aspirate, self.aspirate,
+                   'before', None, None, self, c_vol, location, rate)
```
Comment (amitlissack) [inline api/src/opentrons/protocols/context/protocol_api/instrument_context.py:114]: do_publish would require a `Broker` which the implementation does not have. It's owned by `InstrumentContext` however.

The goal of such a refactor is to make the legacy implementations the keepers of all that is legacy. That means we would move the Broker, the publish decorators, all the api_version checks, etc into the classes in `opentrons.protocols.context.protocol_api`.

**Review by amitlissack — COMMENTED** [review]
From this PR it looks like your challenges are to make the legacy implementation work. 

As I expressed yesterday and this morning, I don't think it's worth the effort. 

When I created the split of the `InstrumentContext` and its inner `_implementation` I didn't know the impact. So far it seems the only impact is fast simulation.

This PR merged to edge will compromise fast simulation. 

This is where I suggest we branch and build the `InstrumentContext` that works for protocol engine. We already have one that works great for protocol api v2.

My opinion.

**Comment by SyntaxColoring** [comment]
Abandoning this PR in favor of other approaches.

**Pull request closed without merging**